charms/worker/k8s/src/charm.py

```diff
@@ -601,6 +601,43 @@ def _get_scrape_jobs(self):
             log.exception("Failed to get COS token.")
         return []
 
+    @on_error(
+        ops.WaitingStatus("Sharing Cluster Version"),
+        AssertionError,
+    )
+    def _update_kubernetes_version(self):
+        """Update the unit Kubernetes version in the cluster relation."""
+        if not (relation := self.model.get_relation("cluster")):
+            assert False, "Missing cluster integration"  # nosec
+        if version := snap_version("k8s"):
+            relation.data[self.unit]["version"] = version
+
+    @on_error(
+        ops.WaitingStatus("Announcing Kubernetes version"),
+        AssertionError,
+    )
+    def _announce_kubernetes_version(self):
+        """Announce the Kubernetes version to the cluster.
+
+        This method ensures that the Kubernetes version is consistent across the cluster.
+        """
+        peer = self.model.get_relation("cluster")
+        worker = self.model.get_relation("k8s-cluster")
+
+        version = snap_version("k8s")
+        assert version, "k8s-snap is not installed"  # nosec
+
+        for relation in (peer, worker):
+            assert relation, "Missing cluster integration"  # nosec
+            units = (unit for unit in relation.units if unit.name != self.unit.name)
```

Comment:
it's possible to deploy just `k8s` without a worker relation.  I don't think you can assert the relation b/c the worker relation may not exist in normal use

```suggestion
        for relation in (peer, worker):
            if not relation:
                continue
            units = (unit for unit in relation.units if unit.name != self.unit.name) 
```